src/main/scala/com/ubirch/models/AcctEventDAO.scala

```diff
@@ -1,40 +1,58 @@
 package com.ubirch.models
 
-import java.util.UUID
-
 import com.ubirch.services.cluster.ConnectionService
+
 import io.getquill.{ CassandraStreamContext, SnakeCase }
-import javax.inject.Inject
 import monix.reactive.Observable
 
-trait AcctEventRowsQueries extends TablePointer[AcctEventRow] {
+import java.time.LocalDate
+import java.util.UUID
+import javax.inject.Inject
 
+trait AcctEventRowsQueries extends TablePointer[AcctEventRow] {
   import db._
 
   //These represent query descriptions only
 
   implicit val pointingAt: db.SchemaMeta[AcctEventRow] = schemaMeta[AcctEventRow]("acct_events")
 
-  def insertQ(acctEventRow: AcctEventRow): db.Quoted[db.Insert[AcctEventRow]] = quote {
+  def insertQ(acctEventRow: AcctEventRow) = quote {
     query[AcctEventRow].insert(lift(acctEventRow))
   }
 
-  def selectAllQ: db.Quoted[db.EntityQuery[AcctEventRow]] = quote(query[AcctEventRow])
+  def selectAllQ = quote(query[AcctEventRow])
 
-  def byOwnerIdQ(ownerId: UUID): db.Quoted[db.EntityQuery[AcctEventRow]] = quote {
+  def byOwnerIdQ(ownerId: UUID) = quote {
     query[AcctEventRow]
       .filter(_.ownerId == lift(ownerId))
       .map(x => x)
   }
 
+  def byOwnerIdCountQ(ownerId: UUID) = quote { byOwnerIdQ(ownerId).size }
+
   def byOwnerIdAndIdentityIdQ(ownerId: UUID, identityId: UUID): db.Quoted[db.EntityQuery[AcctEventRow]] = quote {
     query[AcctEventRow]
       .filter(_.ownerId == lift(ownerId))
       .filter(_.identityId == lift(identityId))
       .map(x => x)
   }
 
-  def deleteQ(ownerId: UUID, acctEventId: UUID): db.Quoted[db.Delete[AcctEventRow]] = quote {
+  def byOwnerIdAndIdentityIdCountQ(ownerId: UUID, identityId: UUID) = quote { byOwnerIdAndIdentityIdQ(ownerId, identityId).size }
+
+  def byOwnerIdAndIdentityIdQ(ownerId: UUID, identityId: UUID, start: LocalDate, end: LocalDate) = quote {
+    query[AcctEventRow]
+      .filter(_.ownerId == lift(ownerId))
+      .filter(_.identityId == lift(identityId))
+      .filter(_.day >= lift(start))
+      .filter(_.day <= lift(end))
```

Comment:
I think the end date should not include to the time range but I dont know the business decision about this actually

Reply:
Thanks. We can discuss it later.